src/Request.hs

```diff
@@ -77,7 +77,15 @@ mkValidator contractInfo@ContractInfo{..} dat lenderTn ctx = validate
     valueToBorrower = valuePaidTo (U.info ctx) (unPaymentPubKeyHash $ borrowersPkh dat)
 
     borrowerGetsWhatHeWants :: Bool
-    borrowerGetsWhatHeWants = assetClassValueOf valueToBorrower (loan dat) >= loanamnt dat
+    borrowerGetsWhatHeWants = assetClassValueOf valueToBorrower (loan dat) == loanamnt dat
+
+    ownHashFilter :: Maybe ValidatorHash -> Bool
+    ownHashFilter mvh = Just (ownHash ctx) == mvh
+      Just vh -> vh == ownHash ctx
```

Comment:
@tomazvila I meant that you delete the two lines 84 and 85 (and keep 83), now it doesn't compile.